Python/pytime.c

```diff
@@ -781,35 +869,71 @@ _PyTime_GetMonotonicClockWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
 
 
 #ifdef MS_WINDOWS
+static int
+init_win_perf_counter(LONGLONG *t0, LONGLONG *perf_frequency,
+                      _PyTime_t *num_p, _PyTime_t *den_p)
+{
+    LARGE_INTEGER now;
+    LARGE_INTEGER freq;
+    _PyTime_t num, den, gcd;
+
+    QueryPerformanceCounter(&now);
+    if (!QueryPerformanceFrequency(&freq)) {
+        PyErr_SetFromWindowsErr(0);
+        return -1;
+    }
+    if (freq.QuadPart < 1) {
+        return -1;
+    }
+
+    *t0 = now.QuadPart;
+    *perf_frequency = freq.QuadPart;
+
+    /* Compute the greatest common divisor to minimize numerator and
+       denominator, to reduce the risk of integer overflow in
+       _PyTime_GetWinPerfCounterWithInfo(). */
+    num = SEC_TO_NS;
+    den = freq.QuadPart;
+    gcd = _PyTime_GCD(num, den);
+    num /= gcd;
+    den /= gcd;
+    if (num < 1 || den < 1) {
```

Comment:
When this is possible?

Reply:
freq.QuadPart is supposed to be >= 1... but I prefer to be super safe, and check anything, "just in case".

The check is cheap and only done once, whereas a crash in bug can be very annoying and complex to debug.